### ISSUE TITLE
Extract into function assist sometimes strips comments

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#13621]
I tried:

```rust
pub fn hello() {
    for _ in 0..1 {
        // Extract the following condition with this comment
        if true {
            continue;
        }

        if false {
            break;
        }
    }
}
```

and got

```rust
use std::ops::ControlFlow;

pub fn hello() {
    for _ in 0..1 {
        if let ControlFlow::Break(_) = fun_name() {
            continue;
        }

        if false {
            break;
        }
    }
}

fn fun_name() -> ControlFlow<()> {
    if true {
        return ControlFlow::Break(());
    }
    ControlFlow::Continue(())
}
```

when it should have been

```rust
use std::ops::ControlFlow;

pub fn hello() {
    for _ in 0..1 {
        if let ControlFlow::Break(_) = fun_name() {
            continue;
        }

        if false {
            break;
        }
    }
}

fn fun_name() -> ControlFlow<()> {
    // Extract the following condition with this comment
    if true {
        return ControlFlow::Break(());
    }
    ControlFlow::Continue(())
}
```

See also #13620 which is potentially related.

**rust-analyzer version**: `rust-analyzer version: 0.4.1286-standalone`

**rustc version**: `rustc 1.65.0 (897e37553 2022-11-02)`

**relevant settings**: -

**Comment by jplatte** [comment]
Here's a failing test case: #13692

**Comment by feniljain** [comment]
Makes sense, thanks for the clarification! :bow:

**Comment by jplatte** [comment]
The previously failing test was included on #13746 and passes now, so I'll close this.
